lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -418,74 +418,72 @@ def add_exemption(post, options = {})
-        return unless options[:stored_credential] && !options[:stored_credential].values.all?(&:nil?)
+        stored_credential = options[:stored_credential]
+        return unless stored_credential && !stored_credential.values.all?(&:nil?)
 
         post[:payment_method_options] ||= {}
         post[:payment_method_options][:card] ||= {}
-        add_stored_credential_transaction_type(post, options) if options[:stored_credential_transaction_type]
-
-        stored_credential = options[:stored_credential]
         post[:payment_method_options][:card][:mit_exemption] = {}
+        card_options = post[:payment_method_options][:card]
 
         # Stripe PI accepts network_transaction_id and ds_transaction_id via mit field under card.
         # The network_transaction_id can be sent in nested under stored credentials OR as its own field (add_ntid handles when it is sent in on its own)
         # If it is sent is as its own field AND under stored credentials, the value sent under its own field is what will send.
-        post[:payment_method_options][:card][:mit_exemption][:ds_transaction_id] = stored_credential[:ds_transaction_id] if stored_credential[:ds_transaction_id]
-        post[:payment_method_options][:card][:mit_exemption][:network_transaction_id] = stored_credential[:network_transaction_id] if stored_credential[:network_transaction_id]
-      end
+        card_options[:mit_exemption][:ds_transaction_id] = stored_credential[:ds_transaction_id] if stored_credential[:ds_transaction_id]
+        card_options[:mit_exemption][:network_transaction_id] = stored_credential[:network_transaction_id] if stored_credential[:network_transaction_id]
 
-      def add_stored_credential_transaction_type(post, options = {})
-        stored_credential = options[:stored_credential]
-        # Do not add anything unless these are present.
-        return unless stored_credential[:reason_type] && stored_credential[:initiator]
+        return unless options[:stored_credential_transaction_type] && %i(reason_type initiator).all? { |k| stored_credential.key?(k) }
 
         # Not compatible with off_session parameter.
         options.delete(:off_session)
-        if stored_credential[:initial_transaction]
-          # Initial transactions must by CIT
-          return unless stored_credential[:initiator] == 'cardholder'
 
-          initial_transaction_stored_credential(post, stored_credential[:reason_type])
-        else
-          # Subsequent transaction
-          subsequent_transaction_stored_credential(post, stored_credential[:initiator], stored_credential[:reason_type])
-        end
+        stored_credential_type = if stored_credential[:initial_transaction] && stored_credential[:initiator] == 'cardholder'
+                                   initial_transaction_stored_credential(post, stored_credential)
+                                 else
+                                   subsequent_transaction_stored_credential(post, stored_credential)
```

Comment:
I'm not sure I'm following but do you mean [this line was removed in my changes](https://github.com/activemerchant/active_merchant/pull/4828/files#diff-19492101b546175aa3ca6a16ec7d0a610d0adae03cbaf130eff54092dec8e0fbL440). I was thinking yesterday just to go ahead an remove all the extra code change I did and just got back to how it was before. Of course just adding the NTID changes.

Reply:
Nope, I was talking about this [line](https://github.com/activemerchant/active_merchant/pull/4828/files#diff-19492101b546175aa3ca6a16ec7d0a610d0adae03cbaf130eff54092dec8e0fbL446)

Reply:
Does [this line](https://github.com/activemerchant/active_merchant/pull/4828/files#diff-19492101b546175aa3ca6a16ec7d0a610d0adae03cbaf130eff54092dec8e0fbR440) not do the same thing?

Reply:
I don't think so because if `initial_transaction ` is true and stored_credential[:initiator] == 'merchant' it will go into the else statement rather than return.

Reply:
Oh I see what you're saying, let me fix that. So we only set `:stored_credential_transaction_type` for `stored_credential[:initiator] = 'cardholder'`?

Reply:
No, if `initial_transaction == true` then `stored_credential[:initiator]` must be set to `'cardholder'` in order to set the `stored_credential_transaction_type`. There are some valid combinations for `stored_credential[:initiator]` when set to `'merchant'`. 

Sorry if I'm explaining this bad, it's just a tangled web of logic. I can write all the scenarios if that's helpful!

Reply:
I get it, I realized what you were saying after I read your earlier comment again. It can be confusing.